tests/nodeos_read_terminate_at_block_test.py

```diff
@@ -0,0 +1,316 @@
+                skip=True
+                continue
+
+            if swapValue is None:
+                cmdArr.append(i)
+            else:
+                cmdArr.append(swapValue)
+                swapValue=None
+
+            if i in toAddOrSwap:
+                swapValue=toAddOrSwap[i]
+                del toAddOrSwap[i]
+        for k,v in toAddOrSwap.items():
+            cmdArr.append(k)
+            cmdArr.append(v)
+        myCmd=" ".join(cmdArr)
+
+    cmd=myCmd + ("" if chainArg is None else (" " + chainArg))
+    node.launchCmd(cmd, cachePopen)
+
+    def isNodeAlive():
+        """wait for node to be responsive."""
+        try:
+            return True if node.checkPulse() else False
+        except (TypeError) as _:
+            pass
+        return False
+
+    isAlive=Utils.waitForTruth(isNodeAlive, timeout, sleepTime=1)
+    if isAlive:
+        Utils.Print("Node relaunch was successful.")
+    else:
+        Utils.Print("ERROR: Node relaunch Failed.")
+        # Ensure the node process is really killed
+        if node.popenProc:
+            node.popenProc.send_signal(signal.SIGTERM)
+            node.popenProc.wait()
+        node.pid=None
+        return False
+
+    node.cmd=cmd
+    node.killed=False
+    return True
+
+def relaunchNode(node: Node, chainArg="", addSwapFlags=None, relaunchAssertMessage="Fail to relaunch"):
+   isRelaunchSuccess = node.relaunch(chainArg=chainArg, addSwapFlags=addSwapFlags, timeout=relaunchTimeout, cachePopen=True)
+   time.sleep(1) # Give a second to replay or resync if needed
+   assert isRelaunchSuccess, relaunchAssertMessage
+   return isRelaunchSuccess
+
+def relaunchTestNode(node: Node, chainArg="", addSwapFlags=None, relaunchAssertMessage="Fail to relaunch"):
+   isRelaunchSuccess = nodeRelaunch(node=node, chainArg=chainArg, addSwapFlags=addSwapFlags, timeout=relaunchTimeout, cachePopen=True)
+   time.sleep(1) # Give a second to replay or resync if needed
+   assert isRelaunchSuccess, relaunchAssertMessage
+   return isRelaunchSuccess
+
+# List to contain the test result message
+testResultMsgs = []
+testSuccessful = True
+try:
+   # Kill any existing instances and launch cluster
+   TestHelper.printSystemInfo("BEGIN")
+   cluster.killall(allInstances=killAll)
+   cluster.cleanup()
+   cluster.launch(
+      prodCount=numOfProducers,
+      totalProducers=numOfProducers,
+      totalNodes=totalNodes,
+      pnodes=1,
+      useBiosBootFile=False,
+      topo="mesh",
+      specificExtraNodeosArgs={
+         0:"--enable-stale-production",
+         4:"--read-mode irreversible",
+         6:"--read-mode irreversible",
+         9:"--plugin eosio::producer_api_plugin"})
+
+   producingNodeId = 0
+   producingNode = cluster.getNode(producingNodeId)
+
+   def stopProdNode():
+      if not producingNode.killed:
+         producingNode.kill(signal.SIGTERM)
+
+   def startProdNode():
+      if producingNode.killed:
```

Comment:
if you expect it not always to be killed, add warning there or other message or assert, depending on your needs.